src/Smalot/PdfParser/RawData/RawDataParser.php

```diff
@@ -723,15 +719,12 @@ protected function getRawObject(string $pdfData, int $offset = 0): array
                     // hexadecimal string object
                     $objtype = $char;
                     ++$offset;
-                    $pregResult = preg_match(
-                        '/^([0-9A-Fa-f\x09\x0a\x0c\x0d\x20]+)>/iU',
-                        substr($pdfData, $offset),
-                        $matches
-                    );
-                    if (('<' == $char) && 1 == $pregResult) {
+
+                    $span = strspn($pdfData, "0123456789abcdefABCDEF\x09\x0a\x0c\x0d\x20", $offset);
+                    if ('<' == $char && $span > 0 && '>' == @$pdfData[$offset + $span]) {
```

Comment:
Your code is fine, but I would like to make it a bit more explicit. This way one is less likely to overlook the `@` in the if clause in my opinion.

```suggestion
                    $dataToCheck = $pdfData[$offset + $span] ?? null;
                    if ('<' == $char && $span > 0 && '>' == $dataToCheck) {
```